Sources/Logging/Logging.swift

```diff
@@ -138,7 +139,23 @@ extension Logging.Level: Comparable {
     }
 }
 
-extension Logging.Metadata.Value: Equatable {}
+extension Logging.Metadata.Value: Equatable {
+    public static func ==(lhs: Logging.Metadata.Value, rhs: Logging.Metadata.Value) -> Bool {
+        switch (lhs, rhs) {
+        case (.string(let lhs), .string(let rhs)):
+            return lhs == rhs
+        case (.stringConvertible(let lhs), .stringConvertible(let rhs)):
+            return lhs.description == rhs.description
+        case (.array(let lhs), .array(let rhs)):
+            return lhs == rhs
+        case (.dictionary(let lhs), .dictionary(let rhs)):
+            return lhs == rhs
+        default:
+            return false
+        }
+    }
```

Comment:
would a private extension to CustomStringConvertible help instead of doing this?

Reply:
Sadly not, we can't add conformance to CustomStringConvertible like that:

```
/Users/ktoso/code/swift-server-logging-api-proposal/Sources/Logging/Logging.swift:142:9: error: extension of protocol 'CustomStringConvertible' cannot have an inheritance clause
extension CustomStringConvertible: Equatable {
^                                  ~~~~~~~~~
```

We also can't do a private extension that adds a conformance, so that's not really an option.

The code here should be generally fine though, I don't suspect it being slower than the generated code, though we can make sure.